File: exif/src/main/java/it/sephiroth/android/library/exif2/ExifUtil.java
package it.sephiroth.android.library.exif2;

import java.text.DecimalFormat;
import java.text.NumberFormat;

/**
 * Created by alessandro on 20/04/14.
 */
public class ExifUtil {

	static final NumberFormat formatter = DecimalFormat.getInstance();

	public static String processLensSpecifications( Rational[] values ) {
		Rational min_focal = values[0];
		Rational max_focal = values[1];
		Rational min_f = values[2];
		Rational max_f = values[3];

		formatter.setMaximumFractionDigits(1);

		StringBuilder sb = new StringBuilder();
		sb.append( formatter.format( min_focal.toDouble() ) );
		sb.append( "-" );
		sb.append( formatter.format( max_focal.toDouble() ) );
		sb.append( "mm f/" );
		sb.append( formatter.format( min_f.toDouble() ) );
		sb.append( "-" );
		sb.append( formatter.format( max_f.toDouble() ) );

		return sb.toString();
	}

}

File: exif/src/main/java/it/sephiroth/android/library/exif2/ExifUtil.kt
package it.sephiroth.android.library.exif2

import java.text.DecimalFormat

/**
 * Created by alessandro on 20/04/14.
 */
object ExifUtil {
	
	private val formatter = DecimalFormat.getInstance()
	
	fun processLensSpecifications(values : Array<Rational>) : String {
		val min_focal = values[0]
		val max_focal = values[1]
		val min_f = values[2]
		val max_f = values[3]
		
		formatter.maximumFractionDigits = 1
		
		val sb = StringBuilder()
		sb.append(formatter.format(min_focal.toDouble()))
		sb.append("-")
		sb.append(formatter.format(max_focal.toDouble()))
		sb.append("mm f/")
		sb.append(formatter.format(min_f.toDouble()))
		sb.append("-")
		sb.append(formatter.format(max_f.toDouble()))
		
		return sb.toString()
	}
	
}
